File: backend/app/schemas/statistics.py
from pydantic import BaseModel


class ChapterWritingStat(BaseModel):
    chapter_id: int
    title: str
    word_count: int


class ProjectStatisticsResponse(BaseModel):
    total_words: int
    chapter_count: int
    average_words_per_chapter: float

    daily_word_goal: int
    daily_word_progress: int
    daily_goal_percentage: float

    chapters: list[ChapterWritingStat]

File: frontend/src/types/statistics.ts
export interface ChapterWritingStat {
  chapter_id: number
  title: string
  word_count: number
}

export interface ProjectStatistics {
  total_words: number
  chapter_count: number
  average_words_per_chapter: number
  daily_word_goal: number
  daily_word_progress: number
  daily_goal_percentage: number
  chapters: ChapterWritingStat[]
}